packages/web3/src/v3/WalletLink.ts

```diff
@@ -42,11 +44,7 @@ export class WalletLink {
 
     private async assertAlreadyLinked(rootKey: ethers.Signer, walletAddress: string) {
         const rootKeyAddress = await rootKey.getAddress()
-        const isLinkedAlready = await this.walletLinkShim.read.checkIfLinked(
-            rootKeyAddress,
-            walletAddress,
-        )
-        if (!isLinkedAlready) {
+        if (!(await this.isLinkedAlready(walletAddress))) {
```

Comment:
So this check no longer asserts that the wallet is linked to the specified root key, is that important?

Reply:
@clemire thanks for this note.
Correct, I don't think we need to check if a wallet is specifically linked to the root key - we could just check if it's already linked to _any_ wallet (b/c the tx will ultimately fail if this is the case).
I realized the `rootKey` param was still being passed around and is unnecessary, so I updated my PR to remove it where applicable.